devito/ir/iet/visitors.py

```diff
@@ -1082,83 +1107,68 @@ def default_retval(cls):
                      appears.
     """
 
-    rules = {
+    rules: dict[str, Callable[[type, Node], bool]] = {
         'type': lambda match, o: isinstance(o, match),
         'scope': lambda match, o: match in flatten(o.children)
     }
 
-    def __init__(self, match, mode='type'):
+    def __init__(self, match: type, mode: str = 'type'):
         super().__init__()
         self.match = match
         self.rule = self.rules[mode]
 
-    def visit_object(self, o, ret=None):
-        return ret
-
-    def visit_tuple(self, o, ret=None):
+    def visit_tuple(self, o: Sequence[Any]) -> Iterator[Any]:
```

Comment:
This appears in both `FindNodes` and `FindSymbols`. Is it worth moving it up a class?

Reply:
Ditto with `visit_list = visit_tuple`